Sources/Toast/Toast.swift

```diff
@@ -0,0 +1,163 @@
+import Foundation
+import SwiftUI
+
+extension Warp {
+    private static let toastCornerRadius: Double = 8
+
+    /**
+        A toast is a message, which provides quick, at-a-glance feedback on the outcome of an action.
+
+        **When to use**
+
+        The Toast is best used for short success, warning, and error messages to confirm an action.
+
+         To use the iOS `Toast` View you need to provide it with
+        - a `ToastStyle` (can be `.error`, `.success` or `.warning`)
+        - a `title`
+        - a `ToastEdge` (can be `.bottom` or `.top`)
+        - a Binding to a boolean value allowing the `Toast` to dismiss
+     */
+    public struct Toast: View {
+
+        /// Preferred style of toast
+        let style: ToastStyle
+
+        /// Text that will be shown in toast
+        let title: String
+
+        /// Edge from where the toast is presented
+        let toastEdge: Warp.ToastEdge
+
+        /// Binding to a boolean value that allows the toast to control dismissal
+        @Binding public var isPresented: Bool
+
+        /// Object responsible for providing colors in different environments and variants.
+        let colorProvider: ColorProvider
+
+        /**
+         - Parameter style: The `ToastStyle` of the `Toast`
+         - Parameter title: Text to display in the `Toast`
+         - Parameter toastEdge: The `ToastEdge` on where to present the `Toast`
+         - Parameter isPresented: Is the `Toast` presented or not
+         - Parameter colorProvider: ColorProvider used for styling the `Toast`, default value is read from `Config`
+         */
+        public init(
+            style: Warp.ToastStyle,
+            title: String,
+            toastEdge: Warp.ToastEdge,
+            isPresented: Binding<Bool>,
+            colorProvider: ColorProvider = Config.colorProvider
+        ) {
+            self.style = style
+            self.title = title
+            self.toastEdge = toastEdge
+            self._isPresented = isPresented
+            self.colorProvider = colorProvider
+        }
+
+        public var body: some View {
+            contentView
+            .overlay(
+                RoundedRectangle(cornerRadius: toastCornerRadius)
+                    .stroke(style.subtleBorderColor(from: colorProvider), lineWidth: 3)
+            )
+            .frame(maxWidth: .infinity)
+            .background(style.backgroundColor(from: colorProvider))
+            .cornerRadius(toastCornerRadius)
+            .transition(AnyTransition.move(edge: toastEdge.asEdge).combined(with: .opacity))
+            .onTapGesture {
```

Comment:
TapGesture is added to the whole background, is that intentional? We don't want that only on the close button? There is no mention of that on Figma. I have asked our designers and they said there shouldn't be a close button there at all and it should auto dismiss because of a11y reasons. They will look into it, so we do not need to do anything regarding this right now. We will fix that later on.

Reply:
> TapGesture is added to the whole background, is that intentional?

It actually was intentional 😄 

I like that you don't have to try to hit a tiny close button to dismiss but can just tap wherever on the view to dismiss. In the ideal world I'd probably also added a swipe gesture recognizer so you could swipe up to dismiss but that may be going overboard 😅 

> They will look into it, so we do not need to do anything regarding this right now. We will fix that later on.

OK we'll come back here and add a timer it seems 😄